elastic/kibana
# [Alerting] Tracking number of alerts in event log `execute` document and adding telemetry for it.

Resolves https://github.com/elastic/kibana/issues/122535

## Summary

* Adds the following fields to the execution metrics of the event log `execute` document for alerting:
  * `number_of_active_alerts`
  * `number_of_new_alerts`
  * `number_of_recovered_alerts`
  * `total_number_of_alerts`
* Adds telemetry to calculate the p50/p90/p99 values for the total number of alerts generated per rule execution, across all rules and broken down by rule type

## To Verify
* Update the alerting telemetry task to run more frequently:

```
--- a/x-pack/plugins/alerting/server/usage/task.ts
+++ b/x-pack/plugins/alerting/server/usage/task.ts
@@ -139,7 +139,7 @@ export function telemetryTaskRunner(
                   avg_execution_time_per_day: dailyExecutionCounts.avgExecutionTime,
                   avg_execution_time_by_type_per_day: dailyExecutionCounts.avgExecutionTimeByType,
                 },
-                runAt: getNextMidnight(),
+                runAt: new Date(), // getNextMidnight(),
               };
             }
           )
```
* Create a rule that will create alerts and let it run
* Verify that the `execute` documents in the event log for this rule have the new fields enumerated above under `kibana.alert.rule.execution.metrics`.
* Navigate to `https://localhost:5601/api/stats?extended=true&legacy=true` and see the following fields inside alerting telemetry:
  * `alerts.percentile_num_alerts_per_day`
  * `alerts.percentile_num_alerts_by_type_per_day `


### Checklist
